packages/core/src/stateUtils.ts

```diff
@@ -1586,6 +1604,15 @@ function resolveAndExecuteActionsWithContext(
     }
 
     if ('execute' in builtinAction) {
+      actorScope.actionExecutor(
+        {
+          type: builtinAction.type,
+          info: actionArgs,
+          params,
+          exec: () => {} // noop
+        },
+        actorScope.self
+      );
       if (actorScope.self._processingStatus === ProcessingStatus.Running) {
         builtinAction.execute(actorScope, params);
```

Comment:
what happens here is beyond convoluted:

1. a `noop` exec gets passed into `actorScope.actionExecutor`
2. then that gets mutated internally there for some actions (raise and sendTo)
3. but then we can also learn that `actorScope.actionExecutor` isn't entirely responsible for the actions' execution because we continue to call `builtinAction.execute` **here**

It's very confusing and hard to reason about as the logic is spread across those 2 places weirdly